modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Slipweakening.md

```diff
@@ -0,0 +1,640 @@
+
+\begin{equation}
+\begin{aligned}
+\begin{array}{r} \int_{V}^{}{\sigma \cdot \nabla\psi}\ dV(2) \end{array}
+\end{aligned}
+\end{equation}
+
+```InertiaForce``` gives the inertia term in the weak form:
+
+\begin{equation}
+\begin{aligned}
+\begin{array}{r} \int_{V}^{}{\rho\ddot{u}\ \psi}\ dV(3) \end{array}
+\end{aligned}
+\end{equation}
+
+The input file defining both kernels are given below, note in 2D formulation, “plane strain” needs to be set:
+
+!listing moose/modules/tensor_mechanics/examples/slip_weakening/2D_slipweakening/tpv2052D.i 
+         block=Modules
+         id=input-block-1
+         caption=StressDivergenceTensors Kernels: Input File (2D)
+
+!listing moose/modules/tensor_mechanics/examples/slip_weakening/3D_slipweakening/tpv2053D.i
+         block=Modules
+         id=input-block-2
+         caption=StressDivergenceTensors Kernels: Input File (3D)
+
+The inertia force kernel is given as follows, with the assumption of small strain, we set ```use_displaced_mesh``` to ```false```:
+
+!listing moose/modules/tensor_mechanics/examples/slip_weakening/3D_slipweakening/tpv2053D.i
+         block=Kernels
+         id=input-block-3
+         caption=InertiaForce Kernels: Input File
+
+#### Custom Kernel: StiffPropDamping
+
+```StiffPropDamping``` is a custom kernel for adding stiffness proportional damping into the system to reduce the high-frequency oscillations. The weak form is expressed as follows:
+
+\begin{equation}
+\begin{aligned}
+\begin{array}{r} q\int_{V}^{}{\overline{\sigma} \cdot \nabla\psi}dV (4) \end{array}
+\end{aligned}
+\end{equation}
+
+Where $\overline{\sigma}$ is the damping stress tensor, $q$ is damping constant. To see how the damping term is introduced, the .h and .cpp file is provided below:
+
+The header file explains its inherence relation with its parent class ```StressDivergenceTensors```, which is introduced earlier.
+
+!listing moose/modules/tensor_mechanics/test/include/kernels/StiffPropDamping.h
+caption=StiffPropDamping: Input File
+
+The source file implements the weak form evaluation at each quadrature point, here we follow similar definition of damping stress tensor given in [!cite](Day_Dalguer_Lapusta_Liu_2005) , Appendix A8:
+
+\begin{equation}
+\begin{aligned}
+\begin{array}{r} \overline{\sigma} = \Delta t\left\lbrack \frac{\sigma_{t} - \sigma_{t - \Delta t}}{\Delta t} \right\rbrack = \left\lbrack \sigma_{t} - \sigma_{t - \Delta t} \right\rbrack (5)\end{array}
+\end{aligned}
+\end{equation}
+
+Where $\sigma_{t}$ and $\sigma_{t - \Delta t}$ are stress tensor from current/last time step. The code snippet is given below, notice that for each custom function, user needs to register the function to their own app using ```registerMooseObject```. (```TensorMechanicsApp``` is a custom app name that can be replaced).
+
+!listing moose/modules/tensor_mechanics/test/include/kernels/StiffPropDamping.C
```

Comment:
@chunhuizhao478 Before this can be reviewed, the documentation needs to be built so that I can get some details on what you are implementing.  You are currently getting this error in the documentation testing:
``` 
/data/civet0/build/moose/modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Slipweakening.md:110
   ┌───────────────────────────────────────────────────────────────────────────────────────────────┐
110│                                                                                               │
111│                                                                                               │
112│!listing moose/modules/tensor_mechanics/test/include/kernels/StiffPropDamping.C                │
113│caption=StiffPropDamping: Source File                                                          │
   └───────────────────────────────────────────────────────────────────────────────────────────────┘
Traceback (most recent call last):
```
I think you need something like this here:
```suggestion
!listing modules/tensor_mechanics/test/include/kernels/StiffPropDamping.C
```

make sure you can build the documentation without any errors and then push it again and let me know so that I can activate the tests.